server/rosetta/client_offline.go

```diff
@@ -0,0 +1,208 @@
+package rosetta
+
+import (
+	"context"
+	"encoding/hex"
+	"strings"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/coinbase/rosetta-sdk-go/types"
+	crgerrs "github.com/tendermint/cosmos-rosetta-gateway/errors"
+	"github.com/tendermint/tendermint/crypto"
+
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+func (c *Client) OperationStatuses() []*types.OperationStatus {
+	return []*types.OperationStatus{
+		{
+			Status:     StatusSuccess,
+			Successful: true,
+		},
+	}
+}
+
+func (c *Client) Version() string {
+	const version = "cosmos-sdk:v0.40.0-rc5/tendermint:0.34.0"
+	return version
+}
+
+func (c *Client) SupportedOperations() []string {
+	var supportedOperations []string
+	for _, ii := range c.ir.ListImplementations("cosmos.base.v1beta1.Msg") {
+		resolve, err := c.ir.Resolve(ii)
+		if err == nil {
+			if _, ok := resolve.(Msg); ok {
+				supportedOperations = append(supportedOperations, strings.TrimLeft(ii, "/"))
+			}
+		}
+	}
+
+	supportedOperations = append(supportedOperations, OperationFee)
+
+	return supportedOperations
+}
+
+func (c *Client) SignedTx(ctx context.Context, txBytes []byte, signatures []*types.Signature) (signedTxBytes []byte, err error) {
+	TxConfig := c.getTxConfig()
+	rawTx, err := TxConfig.TxDecoder()(txBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	txBldr, _ := TxConfig.WrapTxBuilder(rawTx)
+
+	var sigs = make([]signing.SignatureV2, len(signatures))
+	for i, signature := range signatures {
+		if signature.PublicKey.CurveType != "secp256k1" {
+			return nil, crgerrs.ErrUnsupportedCurve
+		}
+
+		cmp, err := btcec.ParsePubKey(signature.PublicKey.Bytes, btcec.S256())
+		if err != nil {
+			return nil, err
+		}
+
+		compressedPublicKey := make([]byte, secp256k1.PubKeySize)
+		copy(compressedPublicKey, cmp.SerializeCompressed())
+		pubKey := &secp256k1.PubKey{Key: compressedPublicKey}
+
+		accountInfo, err := c.accountInfo(ctx, sdk.AccAddress(pubKey.Address()).String(), nil)
+		if err != nil {
+			return nil, err
+		}
+
+		sig := signing.SignatureV2{
+			PubKey: pubKey,
+			Data: &signing.SingleSignatureData{
+				SignMode:  signing.SignMode_SIGN_MODE_LEGACY_AMINO_JSON,
```

Comment:
I haven't worked on the construction part, @sahith-narahari can probably explain better, but IIRC it's because we cannot support anything else but legacy mode atm.